drivers/storage/portworx/deployment_test.go

```diff
@@ -242,6 +243,43 @@ func TestPodSpecWithPortworxContainerResources(t *testing.T) {
 	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
 	assert.Len(t, podSpec.Containers, 1)
 	assert.Equal(t, resources, podSpec.Containers[0].Resources)
+
+	// Memory limit and CPU limit
+	cluster.Spec.Resources = &v1.ResourceRequirements{
+		Requests: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceMemory: resource.MustParse("4Gi"),
+			v1.ResourceCPU:    resource.MustParse("400m"),
+		},
+		Limits: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceMemory: resource.MustParse("8Gi"),
+			v1.ResourceCPU:    resource.MustParse("800m"),
+		},
+	}
+
+	podSpec, err = driver.GetStoragePodSpec(cluster, nodeName)
+	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
+	assert.Len(t, podSpec.Containers, 1)
+	assert.Equal(t, *cluster.Spec.Resources, podSpec.Containers[0].Resources)
+	require.True(t, strings.Contains(strings.Join(podSpec.Containers[0].Args, " "), "--cpus 0.8"))
+	require.True(t, strings.Contains(strings.Join(podSpec.Containers[0].Args, " "), "--memory 8589934592"))
+
+	cluster.Spec.Resources = &v1.ResourceRequirements{
+		Requests: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceMemory: resource.MustParse("4Gi"),
+			v1.ResourceCPU:    resource.MustParse("4"),
+		},
+		Limits: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceMemory: resource.MustParse("8Gi"),
+			v1.ResourceCPU:    resource.MustParse("8"),
+		},
+	}
+
+	podSpec, err = driver.GetStoragePodSpec(cluster, nodeName)
+	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
+	assert.Len(t, podSpec.Containers, 1)
+	assert.Equal(t, *cluster.Spec.Resources, podSpec.Containers[0].Resources)
+	require.True(t, strings.Contains(strings.Join(podSpec.Containers[0].Args, " "), "--cpus 8"))
+	require.True(t, strings.Contains(strings.Join(podSpec.Containers[0].Args, " "), "--memory 8589934592"))
```

Comment:
What happens if we provide an invalid memory value (like some random text)? That's my concern with the parsing error earlier

Reply:
for memory it wont be set, for cpu actually I have not tried, i can add a unit test to it.

Reply:
Would be good to do yeah

Reply:
actually in unit test you can never parse invalid value, as the quantity is an object. in real cluster, you could edit storageCluster to use invalid memory, however operator will never receive the updated stc due to it's invalid.

E1217 21:07:06.547436       1 reflector.go:140] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:262: Failed to watch *v1.StorageCluster: failed to list *v1.StorageCluster: unable to parse quantity's suffix

Reply:
Ahhhh that makes sense: yeah, then I'm fine without testing that obviously.